test-utils/transaction-factory/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Simple transaction factory which distributes tokens from a master
+//! account to a specified number of newly created accounts.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(not(feature = "std"), feature(alloc))]
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::ops::Mul;
+use std::fmt::Display;
+
+use log::info;
+
+use client::block_builder::api::BlockBuilder;
+use client::runtime_api::ConstructRuntimeApi;
+use consensus_common::{BlockOrigin, ImportBlock, ForkChoiceStrategy, SelectChain};
+use consensus_common::block_import::BlockImport;
+use parity_codec::{Decode, Encode};
+use serde::Serialize;
+use sr_primitives::traits::{As, Block, Header, ProvideRuntimeApi, SimpleArithmetic};
+use substrate_service::{FactoryBlock, FactoryFullConfiguration, FullClient, new_client, ServiceFactory, ComponentClient, FullComponents};
+pub use crate::modes::Mode;
+
+mod modes;
+mod complex_mode;
+mod simple_modes;
+
+pub trait RuntimeAdapter {
+	type AccountId;
+	type Extrinsic;
```

Comment:
Why not just take `Block` as associated type? It contains `Extrinsic`, `Header`, `Hash`, `BlockNumber` etc.